pages/blog/building-async-flight-notification-service.md

```diff
@@ -0,0 +1,257 @@
+
+```yaml
+type: object 
+properties: 
+    carrierCode: 
+        type: string 
+        description: 2 to 3-character IATA carrier code 
+        example: "LH" 
+    flightNumber: 
+        type: integer 
+        minimum: 1 
+        description: 1 to 4-digit number of the flight 
+        example: "193" 
+    scheduledDepartureDate: 
+        type: string 
+        format: date-time 
+        description: scheduled departure date of the flight, local to the departure airport. 
+        example: "2020-10-20" 
+```
+
+A `flightStatus` message to notify about changes. When the service detects a change in flight status, it triggers a notification event to alert the user. The payload of the `flightStatus` message consists of the following structure:
+
+- `flight` and `user` schemas (the same as in the `flightQueue` message) to identify the flight emitting the event and the user receiving the notification.
+
+- Two `segment` schemas corresponding to the origin and destination. This lets us notify about changes to both departure and arrival. 
+
+```yaml
+type: object 
+properties: 
+  iataCode: 
+    type: string 
+    description: 2 to 3-character IATA carrier code 
+    example: "MAD" 
+  scheduledDate: 
+    type: string 
+    format: date-time 
+    description: scheduled datetime of the flight, local to the airport. 
+    example: "2020-10-20 19:15" 
+  gate: 
+    type: string 
+    description: departure gate
+    example: "2D" 
+  terminal: 
+    type: string 
+    description: airport terminal 
+    example: "4" 
+```
+
+Messages are shared among services so it’s important to correctly organize the YAML definition files under a common folder. In our case, we call it common:
+
+        common/
+            messages/
+                flight_queue.yaml
+                flight_status.yaml
+            schemas/
+                flight.yaml
+                segment.yaml
+                user.yaml
+
+Services communicate through channels using the publish/subscribe pattern. Our architecture uses two different channels:
+
+- `flight/queue` to manage and queue the flights to be monitored.
+- `flight/update` to manage the notifications about flight updates.
+
+Each service contains an `asyncapi.yaml` file with the description of the service and server and channel information. Let's take a look to the final `asyncapi.yaml` file of the Subscriber service to see how the messages and channels are organized:
+
+```yaml
+asyncapi: '2.0.0'
+info:
+  title: Flight Subscriber Service
+  version: '1.0.0'
+  description: |
+     Allows users to subscribe events from a given flight
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+servers:
+  development:
+    url: mqtt://localhost:1883
+    protocol: mqtt
+channels:
+  flight/queue:
+    description: |
+      queue flight in order to retrieve status
+    subscribe:
+      summary: Inform about the status of a subscribed flight
```

Comment:
it sounds to me that this operation is there to inform about the status of a subscribed flight. But it is not right? it is about `Receive information about the flight that should be monitored for changes`?